### PR TITLE
[RLlib] Rename RLTrainer to Learner and TrainerRunner to LearnerGroup

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/examples/learner/multi_agent_cartpole_ppo.py [hunk 1]
```diff
@@ -101,7 +101,7 @@ def policy_mapping_fn(agent_id, episode, worker, **kwargs):
         .training(num_sgd_iter=10)
         .multi_agent(policies=policies, policy_mapping_fn=policy_mapping_fn)
         .rl_module(_enable_rl_module_api=True)
-        .training(_enable_rl_trainer_api=True)
+        .training(_enable_learneri=True)
```
Comment (avnishn) [inline rllib/examples/learner/multi_agent_cartpole_ppo.py:104]: is this supposed to be `_enable_learneri` or `_enable_learner`

**Review by avnishn — COMMENTED** [review]
I think I caught a typo that would lead to a bug. Otherwise LGTM

File: rllib/examples/learner/multi_agent_cartpole_ppo.py (hunk 1 above)
Comment (kouroshHakha) [reply rllib/examples/learner/multi_agent_cartpole_ppo.py:104]: nice catch.

Comment (kouroshHakha) [reply rllib/examples/learner/multi_agent_cartpole_ppo.py:104]: the. **kwargs is stupid, it will just pass through the non-existing parameters.

Comment (kouroshHakha) [reply rllib/examples/learner/multi_agent_cartpole_ppo.py:104]: nvm it was caught by the tests :)

**Review by gjoliver — APPROVED** [review]

**Pull request merged**